app/models/spreadsheet_service.rb

```diff
@@ -0,0 +1,25 @@
+require 'google/apis/sheets_v4'
+
+class SpreadsheetService
+  def self.get_data(spreadsheet_id)
+    service = Google::Apis::SheetsV4::SheetsService.new
+    service.key = ENV['GOOGLE_SHEET_API_KEY']
+
+    begin 
+      sheet = service.get_spreadsheet(spreadsheet_id)
+      # A1 notation: range == 'All Cells' 
+      range = sheet.sheets[0].properties.title
+      response = service.get_spreadsheet_values(spreadsheet_id, range)
+    rescue Exception => e
+      raise ActionController::BadRequest.new(), e.message()
+    end
+
+    rows = response.values
+
+    Enumerator.new do |enum|
```

Comment:
`get_data` returns an `Enumerator` to avoid having all rows loaded in memory when the CSV file is being written. @bcardiff, do you think it's ok?

Reply:
If depends on the type of `rows`. If it is an `Array`, there is no real gain to use an `Enumerator`. In the csv is because it was parsed one line at a time.

Reply:
Yes, it's an `Array`